dsfhub/resource_data_source_test.go

```diff
@@ -40,62 +233,186 @@ func TestAccDSFDataSource_basic(t *testing.T) {
-		if rs.Type != dsfDataSourceType {
+		if rs.Type != dsfDataSourceResourceType {
 			continue
 		}
 		return fmt.Sprintf("%s", rs.Primary.ID), nil
 	}
 	return "", fmt.Errorf("error finding DSF dataSourceId")
 }
 
-func testCheckDSFDataSourceExists(dataSourceId string) resource.TestCheckFunc {
-	log.Printf("[INFO] Running test testCheckDSFDataSourceExists \n")
-	return func(state *terraform.State) error {
-		res, ok := state.RootModule().Resources[dataSourceId]
-		if !ok {
-			return fmt.Errorf("DSF Data Source resource not found by dataSourceId: %s", dataSourceId)
-		}
-		serverType, ok := res.Primary.Attributes["server_type"]
-		if !ok || serverType == "" {
-			return fmt.Errorf("DSF Data Source Server Type does not exist for dataSourceId %s", dataSourceId)
-		}
-		client := testAccProvider.Meta().(*Client)
-		_, err := client.ReadDSFDataSource(res.Primary.ID)
-		if err != nil {
-			return fmt.Errorf("DSF Data Source Server Type: %s (dataSourceId: %s) does not exist", serverType, dataSourceId)
-		}
-		return nil
-	}
-}
-
-func testAccCheckDSFDataSourceConfigBasic(t *testing.T) string {
-	log.Printf("[INFO] Running test testAccCheckDSFDataSourceConfigBasic \n")
-	return fmt.Sprintf(`
-resource "%s" "my_test_data_source" {
-	admin_email = "%s"
-	arn = "%s"
-	asset_display_name = "%s"
-	gateway_id = %s
-	server_host_name = "%s"
-	server_type = "%s"
-}`, dsfDataSourceResourceName, testAdminEmail, testArn, testAssetDisplayName, testGatewayId, testServerHostName, testDSServerType)
-}
-
+// Confirm assets are destroyed after an acceptance test run
 func testAccDSFDataSourceDestroy(state *terraform.State) error {
-	log.Printf("[INFO] Running test testAccDSFDataSourceDestroy \n")
+	log.Printf("[INFO] Running test testAccDSFDataSourceDestroy")
+	// allow "disableAsset" playbook enough time to run
+	time.Sleep(5 + time.Second)
+
+	// check if asset still exists on hub
 	client := testAccProvider.Meta().(*Client)
 	for _, res := range state.RootModule().Resources {
-		if res.Type != "dsf_data_source" {
+		if res.Type != dsfDataSourceResourceType {
 			continue
 		}
-		dsfDataSourceId := res.Primary.ID
-		readDSFDataSourceResponse, err := client.ReadDSFDataSource(dsfDataSourceId)
+		assetId := res.Primary.ID
+		readDSFDataSourceResponse, err := client.ReadDSFDataSource(assetId)
 		if readDSFDataSourceResponse.Errors == nil {
-			return fmt.Errorf("DSF Data Source %s should have received an error in the response", dsfDataSourceId)
+			return fmt.Errorf("DSF Data Source %s should have received an error in the response", assetId)
 		}
 		if err == nil {
-			return fmt.Errorf("DSF Data Source %s still exists for gatewayId: %s", dsfDataSourceId, testGatewayId)
+			return fmt.Errorf("DSF Data Source %s still exists", assetId)
 		}
 	}
 	return nil
 }
+
+// Configs
+func testAccDSFDataSourceConfig_Basic(resourceName string, adminEmail string, assetId string, gatewayId string, serverHostName string, serverType string) string {
```

Comment:
it's not necessary now but when we eventually rewrite the provider, it'd be nice to either import these from a test data file and/or rename them something slightly more intuitive to indicate that they're just data, not a test of their own

Reply:
I'm just following [terraform's naming convention](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/testcase) of lowercase `testAcc` for function used by acceptance tests and then `Config_<config description>` to indicate it is test data. Not opposed to changing it though if you have better suggestions.

Reply:
Liked the idea of having these separated out into their own files though, will make that change.